drivers/sdp3x/sdp3x.c

```diff
@@ -0,0 +1,458 @@
+    return 1;
+}
+
+int32_t sdp3x_read_single_temperature(sdp3x_t *dev, uint8_t tempComp)
+{
+    _SDP3x_start_triggered(dev, tempComp);
+#ifndef SDP3X_PARAM_IRQ_PIN
+    /* Wait for measurement to be ready if irq pin not used */
+    xtimer_usleep(DATA_READY_SLEEP_US);
+#else
+    /* Try to lock mutex till the interrupt is raised or till timeut happens */
+    xtimer_mutex_lock_timeout(&dev->mutex, DATA_READY_SLEEP_US);
+#endif
+    return _SDP3x_read_temp(dev);
+}
+
+int32_t sdp3x_read_single_differential_pressure(sdp3x_t *dev,
+                                               uint8_t tempComp)
+{
+    _SDP3x_start_triggered(dev, tempComp);
+#ifndef SDP3X_PARAM_IRQ_PIN
+    /* Wait for measurement to be ready if irq pin not used */
+    xtimer_usleep(DATA_READY_SLEEP_US);
+#else
+    /* Try to lock mutex till the interrupt is raised or till timeut happens */
+    xtimer_mutex_lock_timeout(&dev->mutex, DATA_READY_SLEEP_US);
+#endif
+    return _SDP3x_read_pressure(dev);
+}
+
+int8_t sdp3x_read_single_measurement(sdp3x_t *dev, uint8_t tempComp,
+                                     sdp3x_measurement_t *result)
+{
+    _SDP3x_start_triggered(dev, tempComp);
+#ifndef SDP3X_PARAM_IRQ_PIN
+    /* Wait for measurement to be ready if irq pin not used */
+    xtimer_usleep(DATA_READY_SLEEP_US);
+#else
+    /* Try to lock mutex till the interrupt is raised or till timeut happens */
+    xtimer_mutex_lock_timeout(&dev->mutex, DATA_READY_SLEEP_US);   
+#endif
+    /* read in sensor values here */
+    int16_t data[3];
+    uint8_t ret = _SDP3x_read_data(dev, data);
+    if (ret != 0) {
+        DEBUG("[SDP3x] Error reading data\n");
+        return 0;
+    }
+    else {
+        result->differential_pressure = _SDP3x_convert_to_pascal(dev, data[0]);
+        result->temperature = _SDP3x_convert_to_celsius(data[1]);
+        result->dp_scale_factor = data[2];
+    }
+    return 1;
+}
+
+int8_t  sdp3x_start_continuous(sdp3x_t *dev, uint8_t tempComp,
+                               uint8_t avg)
+{
+    int ret = 0;
+    uint8_t cmd[2] = { 0x36, 0 };
+    if (tempComp == 1) {
+        if (avg == 0) {
+            cmd[1] = 0x08;
+        }
+        else {
+            cmd[1] = 0x03;
+        }
+    }
+    else {
+        if (avg == 0) {
+            cmd[1] = 0x1E;
+        }
+        else {
+            cmd[1] = 0x15;
+        }
+    }
+
+    i2c_acquire(DEV_I2C);
+    ret = i2c_write_bytes(DEV_I2C, DEV_ADDR, cmd, 2, 0);
+    i2c_release(DEV_I2C);
+
+    if (ret != 0) {
+        DEBUG("[SDP3x] Error starting continuous measurement\n");
+    }
+
+    dev->continuous_measurement = true;
+    return ret;
+}
+
+int8_t sdp3x_read_continuous(sdp3x_measurement_t *result, sdp3x_t *dev)
+{
+    /* read in sensor values here */
+    int16_t data[3];
+    uint8_t ret = _SDP3x_read_data(dev, data);
+
+    if (ret != 0) {
+        DEBUG("[SDP3x] Error reading date\n");
+    }
+    else {
+        result->differential_pressure = _SDP3x_convert_to_pascal(dev, data[0]);
```

Comment:
```suggestion
        result->differential_pressure = _SDP3x_convert_to_pascal(dev, data[0], data[2]);
```

Reply:
Yeah that's exactly what I did right now :D

Reply:
That means you can also drop `sdp3x_model_t model` - now that the driver detects the scale factor automatically, there is no need to specify which model of the sensor is used.
The user can just attach either of the two and it will just work without changing the configuration.

Reply:
Yeah true! Will change this as well

Reply:
Or you could just drop that part of the product number check too - it's only one byte that changes.
Since the driver now works with both model numbers, that part of the product number is no longer relevant.
Or you could check if it's either of the two allowed values.

Reply:
I just thought this would also be a good check if the params entered by the user are correct